apps/server/src/modules/authorization/api/authorization-reference.controller.ts

```diff
@@ -0,0 +1,33 @@
+import { Authenticate, CurrentUser, ICurrentUser } from '@modules/authentication';
+import { Body, Controller, InternalServerErrorException, Param, Post, UnauthorizedException } from '@nestjs/common';
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
+import { ApiValidationError } from '@shared/common';
+import { AuthorizationReferenceUc } from './authorization-reference.uc';
+import { AuthorizationBodyParams, AuthorizationUrlParams, AuthorizedReponse } from './dto';
+
+@Authenticate('jwt')
+@ApiTags('Authorization')
+@Controller('authorization')
+export class AuthorizationReferenceController {
+	constructor(private readonly authorizationReferenceUc: AuthorizationReferenceUc) {}
+
+	@ApiResponse({ status: 200, type: AuthorizedReponse })
+	@ApiResponse({ status: 400, type: ApiValidationError })
+	@ApiResponse({ status: 401, type: UnauthorizedException })
+	@ApiResponse({ status: 500, type: InternalServerErrorException })
+	@Post('authorize-by-reference/referenceType/:referenceType/referenceId/:referenceId')
```

Comment:
This is not a POST but a GET request.

Reply:
Warum sollte es ein get sein und wie würdest du das mit dem context vom body lösen?

Reply:
Über das get hatte ich auch nachgedacht, nur scheint mir das relativ schwer erweiterbar wenn sich dort irgendwas ändert und die url nicht noch länger werden soll. Ich hatte sogar überlegt das referenceType und id in den body zu verschieben. 
Man könnte auch sagen das ein AuthorizedContext erstellt wird. Daher bin ich dann auf das post gewechselt.

Reply:
Ich denke, es ist ein GET, weil hier nur Informationen vom Server abgerufen werden. Bei einem POST ist mW die Erwartung, dass eine Änderung auf dem Server stattfindet. 

Ich würde einfach alle nötigen Parameter in der URL übergeben. Ich wüsste jetzt auch erstmal nicht, warum das schwerer zu erweitern sein soll, als wenn die Informationen im Body stehen.